runtime/compiler/runtime/RuntimeAssumptions.cpp

```diff
@@ -90,20 +81,34 @@ TR_PatchNOPedGuardSiteOnClassPreInitialize::hashCode(char *sig, uint32_t sigLen)
    return sum;
    }
 
+void
+TR_PatchNOPedGuardSiteOnClassPreInitialize::reclaim()
+   {
+   if (_key != 0)
```

Comment:
Do we still need this since I think we are no longer nulling `_key` in reclaim the original code should work and might be less confusing - eg how could the key ever be null?

Reply:
Might be worth doing a `TR_ASSERT_FATAL(_key != NULL, ....);`

Reply:
Fixed in [force push](https://github.com/eclipse/openj9/compare/f73530051f1fd15719269f628d2fe20c9065869f..5b57a0b92d691020cd06040de2ce7f1d86a74306).